gcloud-java-compute/src/main/java/com/google/gcloud/compute/License.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud.compute;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.common.base.MoreObjects;
+
+import java.io.Serializable;
+import java.util.Objects;
+
+/**
+ * A Google Compute Engine License. A License represents a software license. Licenses are used to
+ * track software usage in images, persistent disks, snapshots, and virtual machine instances.
+ */
+public final class License implements Serializable {
+
+  private static final long serialVersionUID = 6907923910319640363L;
+
+  private final LicenseId licenseId;
+  private final Boolean chargesUseFee;
+  private final String selfLink;
+
+  License(LicenseId licenseId, Boolean chargesUseFee, String selfLink) {
+    this.licenseId = checkNotNull(licenseId);
+    this.chargesUseFee = chargesUseFee;
+    this.selfLink = selfLink;
+  }
+
+  /**
+   * Returns the identity of the license.
+   */
+  public LicenseId licenseId() {
+    return licenseId;
+  }
+
+  /**
+   * Returns {@code true} if the customer will be charged a license fee for running software that
+   * contains this license on an instance.
+   */
+  public Boolean chargesUseFee() {
+    return chargesUseFee;
+  }
+
+  /**
+   * Returns a service-defined URL for the license.
+   */
+  public String selfLink() {
+    return selfLink;
+  }
+
+  @Override
+  public String toString() {
+    return MoreObjects.toStringHelper(this)
+        .add("licenseId", licenseId)
+        .add("chargesUseFee", chargesUseFee)
+        .add("selfLink", selfLink)
+        .toString();
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(licenseId, chargesUseFee, selfLink);
```

Comment:
To be consistent with the other classes' (eg DiskType, MachineType) hashcode methods, this could just be licenseId, right?